### PR TITLE
Add answer sorting and pagination (close #50 close #51 close #52)

### LINKED ISSUE #50: API: Sort Answers
As a developer I want to have an option to sort the answers in ascending and descending order by the number of votes it received or the date submitted. I also want to have a default sorting so I can fetch the most recent answers by default.

### LINKED ISSUE #51: API: Limit the Number of Answers
As a developer I want to have the option to limit the number of answers I receive when I fetch data from the `answers` endpoint. I also want to have a default limit so I don't fetch every single answer if I don't provide the limit parameter.

### LINKED ISSUE #52: API: Paginate Answers
As as developer I want to have an option to paginate through the answers so I don't have to fetch every answer to display a small portion only.

### PR DESCRIPTION
**Pull request by zsoltime**
Added the sorting and pagination for the answers endpoint. I also made a few changes in `server/index.js` so we can use [supertest](https://www.npmjs.com/package/supertest) to test the endpoints.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -9,13 +9,16 @@ const app = new Koa();
 const router = new Router();
 
 const port = process.env.API_PORT || 3000;
+const db = process.env.NODE_ENV === 'test'
+  ? process.env.DB_TEST
```
Comment (Heyjp) [inline server/index.js]: I like the naming for consistency but can you change the `.env.example` and also the `questions.test.js` mongoose connect call as it now throws errors with your environment variable.

Comment (Heyjp) [inline server/index.js:13]: I like the naming for consistency but can you change the `.env.example` and also `questions.test.js`  the  `mongoose.connect` call now throws errors with your new environment variable.

File: server/controller/answers.js [hunk 2]
```diff
@@ -31,12 +31,40 @@ exports.addAnswer = async (ctx) => {
...
 
   await Answer.find({ question_id })
+    .sort({ [sort_by]: sort })
+    .skip(skip)
+    .limit(limit)
     .then((res) => {
       if (res.length === 0) {
-        ctx.body = { err: 'No Answers with that ID' };
+        // is it really an error?
+        // shouldn't we just return an empty array?
```
Comment (Heyjp) [inline server/controller/answers.js]: Possible we could send a message like, "be first to answer the question" or something a bit more user friendly. Could be handled on the front end if it's an empty array. Thoughts?

**Review by Heyjp — CHANGES_REQUESTED** [review]
Made some comments for minor changes. Otherwise code look good and is ready for merge.

File: server/controller/answers.js (hunk 2 above)
Comment (zsoltime) [reply server/controller/answers.js]: I also thought about returning an empty array so we could handle it on the front end.

Comment (zsoltime) [reply server/controller/answers.js]: Or we can return an empty array if `page` is 1 and an error only if we're not on the first page. Something like `page doesn't exist` should work.

File: server/index.js (hunk 1 above)
Comment (zsoltime) [reply server/index.js:13]: We don't need to touch `.env.example` and `.env` as we only use `NODE_ENV=test` for tests and it's already present in `package.json`. I'll check the `questions.test.js` though.

File: server/controller/answers.js (hunk 2 above)
Comment (Heyjp) [reply server/controller/answers.js]: That's a good idea, go with that.

File: server/index.js (hunk 1 above)
Comment (zsoltime) [reply server/index.js:13]: Doh, you meant `DB_TEST`... I tried to figure out why `TEST_DB` is undefined in your test files. 😂

**Comment by zsoltime** [comment]
@Heyjp Your tests should work again with this fix 😃

**Review by Heyjp — APPROVED** [review]

**Pull request merged**